enclave-manager/server/server.go

```diff
@@ -227,26 +227,23 @@ func (c *WebServer) RunStarlarkPackage(ctx context.Context, req *connect.Request
 		Msg: req.Msg.RunStarlarkPackageArgs,
 	}
 
-	apicStream, err := (*apiContainerServiceClient).RunStarlarkPackage(ctx, serviceRequest)
-	ctxWithCancel, cancel := context.WithCancel(ctx)
-	logs := getRuntimeLogsWhenCreatingEnclave(cancel, apicStream)
-	for {
-		select {
-		case <-ctxWithCancel.Done():
-			err := apicStream.Close()
-			close(logs)
-			if err != nil {
-				return stacktrace.Propagate(err, "Error occurred after closing the stream")
-			}
-			return nil
-		case resp := <-logs:
-			errWhileSending := str.Send(resp)
-			if errWhileSending != nil {
-				logrus.Errorf("error occurred: %+v", errWhileSending)
-				return stacktrace.Propagate(errWhileSending, "Error occurred while sending streams")
-			}
+	apiClient, err := (*apiContainerServiceClient).RunStarlarkPackage(ctx, serviceRequest)
+
+	for apiClient.Receive() {
+		resp := apiClient.Msg()
+		errWhileSending := str.Send(resp)
+		if errWhileSending != nil {
+			logrus.Errorf("Error occurred: %+v", errWhileSending)
+			return stacktrace.Propagate(errWhileSending, "Error occurred while sending streams")
 		}
 	}
+
+	if errWhileReceive := apiClient.Err(); errWhileReceive != nil {
+		logrus.Errorf("Error occurred: %+v", errWhileReceive)
+		return stacktrace.Propagate(errWhileReceive, "Error occurred while receiving data from engine")
+	}
+	ctx.Done()
```

Comment:
what's `ctx.Done()` doing here? From my understanding, [`Context.Done()`](https://pkg.go.dev/context#Context) just returns a channel.

Reply:
oh lol; you are right; will remove this.